server/autobuild

```diff
@@ -406,16 +415,25 @@ os_gentoo() {
 
 handle_options "$@"
 
-os_macos   "$@" || \
-os_freebsd "$@" || \
-os_arch    "$@" || \
-os_centos  "$@" || \
-os_openbsd "$@" || \
-os_fedora  "$@" || \
-os_debian  "$@" || \
-os_gentoo  "$@" || \
-os_msys2   "$@" || \
-os_nixos   "$@" || \
+if [ -n "$OS" ]; then
+    OSES="freebsd arch centos openbsd fedora debain gentoo msys2 nixos"
```

Comment:
Great thanks.  But can we do without this list ?

Reply:
Possibly, I used it to validate user input. I could probably just run `os_$OS` without checking, but that could cause "injection" attacks e.g. someone defines `os_hacks` somewhere and the user accidentally specifies `--os hacks` on the command line. This is probably not a realistic attack, so I could remove the check if you want.

Reply:
Yes, probably not a good idea to just execute some passed word.